Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -538,6 +542,11 @@ void HTMLInputElement::updateType()
     m_inputType = WTFMove(newType);
     m_inputType->createShadowSubtreeIfNeeded();
 
+    if (hasAttributeWithoutSynchronization(readonlyAttr) && oldTypeSupportsReadOnly != newTypeSupportsReadOnly) {
```

Comment:
I think the check should be written as `oldTypeSupportsReadOnly != newTypeSupportsReadOnly && hasAttributeWithoutSynchronization(readonlyAttr)`, since the boolean comparison is cheaper.